tests/Template/TemplateTest.php

```diff
@@ -2230,6 +2230,29 @@ interface B {}
                     '$y' => 'array<array-key, A&B>',
                 ]
             ],
+            'templateEmptyParamCoercion' => [
+                '<?php
+                    namespace NS;
+                    use Countable;
+
+                    /** @template T */
+                    class Collection
+                    {
+                        /** @psalm-var iterable<T> */
+                        private $data;
+
+                        /** @psalm-param iterable<T> $data */
+                        public function __construct(iterable $data = []) {
+                            $this->data = $data;
+                        }
+                    }
+
+                    class Item {}
+                    /** @psalm-param Collection<Item> $c */
+                    function takesCollectionOfItems(Collection $c): void {}
+
+                    takesCollectionOfItems(new Collection());',
```

Comment:
So the idea is that `[]` is implicitly upcast to an `iterable<T>` here?

To make this more readable, I'd use an explicit `[]` at call-time, rather than as default parameter value: that would make the example more readable.

Reply:
I'll add both examples